### PR TITLE
fix: front-end changed to make use of new token fetch on the backend

### PR DESCRIPTION
**Pull request by edewit**
Fixes: #910

### CODE REVIEW

**Comment by edewit** [comment]
@gastaldi to test add the clientId and secret in the [git-providers.yaml](https://github.com/fabric8-launcher/launcher-application/pull/911/files#diff-718f83f35b01afedca738e07077a1260R8-R9)

File: web/src/main/java/io/fabric8/launcher/web/producers/GitServiceProducer.java [hunk 1]
```diff
@@ -61,11 +68,12 @@ GitService getGitService(final HttpServletRequest request,
         GitServiceConfig gitServiceConfig = getGitServiceConfig(request);
         GitServiceFactory gitServiceFactory = gitServiceFactories.getGitServiceFactory(gitServiceConfig.getType());
         String provider = gitServiceFactory.getName().toLowerCase();
-        String gitAuthHeader = request.getHeader(GIT_AUTHORIZATION_HEADER);
+        String header = request.getHeader(GIT_AUTHORIZATION_HEADER);
+        String gitAuthHeader = tokenProvider.decryptToken(removeBearerPrefix(header));
```
Comment (gastaldi) [inline web/src/main/java/io/fabric8/launcher/web/producers/GitServiceProducer.java]: This breaks existing clients that send the Git token directly in the request. I'd prefer to add a new header indicating that the Bearer token needs to be decrypted before using it (something like `X-Git-Authorization-Encrypted:true`)

**Comment by gastaldi** [comment]
Fixes #910

File: git-providers.yaml [hunk 2]
```diff
@@ -3,6 +3,10 @@
   apiUrl: https://api.github.com
   repositoryUrl: https://github.com
   type: GITHUB
+  serverProperties:
+    oauthUrl: https://github.com/login/oauth/access_token
+    clientId:
```
Comment (gastaldi) [inline git-providers.yaml]: I am confused, is thie client ID set as an ENV or here?

**Comment by gastaldi** [comment]
Also don't forget to remove unused ENVs from the template in `openshift/template.yaml` and regenerate the unified template

**Comment by gastaldi** [comment]
Looks like the build failed in the frontend module?

File: web/src/main/java/io/fabric8/launcher/web/producers/GitServiceProducer.java (hunk 1 above)
Comment (edewit) [reply web/src/main/java/io/fabric8/launcher/web/producers/GitServiceProducer.java]: removed this completly

File: git-providers.yaml (hunk 2 above)
Comment (edewit) [reply git-providers.yaml]: client now loads these settings and no longer uses the env vars

**Comment by edewit** [comment]
@gastaldi still need to update the operator to set the values in the `git-provider.yaml`

File: git-providers.yaml [hunk 3]
```diff
@@ -18,5 +23,10 @@
   apiUrl: https://try.gitea.io
   repositoryUrl: https://try.gitea.io
   type: GITEA
+  clientProperties:
+    giteaOAuthUrl: http://[YOUR-GITEA-URL]/login/oauth/authorize
+    redirectUri: http://frontend-location/fixed
   serverProperties:
     adminUser: admin
+    oauthUrl: https://github.com/login/oauth/access_token
```
Comment (gastaldi) [inline git-providers.yaml]: This looks weird, maybe leave empty for now?

Comment (gastaldi) [inline git-providers.yaml]:
```suggestion
    oauthUrl: http://[YOUR-GITEA-URL]/login/oauth/access_token
```

Comment (edewit) [reply git-providers.yaml]: if you leave it empty it's no longer serializable

**Pull request merged**